mpact_api/mpact/views.py

```diff
@@ -0,0 +1,121 @@
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from . import constants
+from .logger import logger
+from .models import ChatData, UserChat, UserData
+from .serializers import UserDataSerializer
+from .services import save_chatdata, save_userchat, save_userdata, set_webhook
+
+
+class Webhook(APIView):
+    """
+    This is only called once for the webhook setup.
+    """
+
+    def get(self, request):
+        webhook = set_webhook()
+        if webhook:
+            return Response("webhook setup ok")
+        else:
+            return Response("webhook setup failed")
+
+
+class ListenMessages(APIView):
+    """
+    Telegram webhook makes a post request.
+    """
+
+    def post(self, request):
+        """
+        request.data {
+           "update_id": 30927953,
+           "message":
+             {
+               "message_id": 51,
+               "from": {"id": 1229986355, "is_bot": False, "first_name": "John", "last_name": "Walter", "language_code": "en"},
+               "chat": {"id": -433354696, "title": "Testing Group", "type": "group", "all_members_are_administrators": True},
+               "date": 1604317469,
+               [
+                   'text': 'Hello, How are you?'
+               ] or
+               [
+                    "new_chat_participant": {"id": 1321604047, "is_bot": True, "first_name": "demo_bot", "username": "test_bot"},
+                    "new_chat_member": {"id": 1321604047, "is_bot": True, "first_name": "demo_bot", "username": "test_bot"},
+                    "new_chat_members": [{"id": 1321604047, "is_bot": True, "first_name": "demo_bot", "username": "test_bot"}]
+               ] or
+               [
+                    'left_chat_member': {'id': 879250491, 'is_bot': False, 'first_name': 'John', 'last_name': 'Walter'}
+               ]
+             }
+         }
+
+        Above square brackets will be based on the event such as text message, new chat member or a member left the group.
+        """
+        message = request.data.get(constants.MESSAGE)
+        logger.debug(message)
+        chat_data = message.get(constants.CHAT)
+        from_data = message.get(constants.FROM)
+
+        if constants.NEW_CHAT_PARTICIPANT in message:
+            # This is used whenever bot or some other members are added to the group
+            user_data = message.get(constants.NEW_CHAT_PARTICIPANT)
```

Comment:
This code would be easier to read and less verbose if you import the constants you use at the top, e.g.
```python
from .constants import USERNAME, # ...
# ...
            user_data = message[NEW_CHAT_PARTICIPANT]
```
(If `message.get()` sets `user_data` to `None`, the next line will raise an `AttributeError`. You might as well drop the `get()`. If the key is missing, it will raise a `KeyError` instead of the `AttributeError`, which could be more useful for diagnosing the problem.)